components/inputs/input-number.js

```diff
@@ -45,9 +62,18 @@ class InputNumber extends LitElement {
 		this.labelHidden = false;
 		this.required = false;
 
+		this._formattedValue = '';
 		this._inputId = getUniqueId();
 	}
 
+	firstUpdated(changedProperties) {
```

Comment:
I actually think you want this code to run in `updated()`, since anytime the `value` is updated you want to re-check it and re-format it. For example (and good tests to add when you have them), running these should both result in the value being re-formatted which today it won't:

```
// setting the HTML attribute
elem.setAttribute('value', '39.2');

// setting the underlying property
elem.value = 39.2;
```

Another approach (which the time input uses) is to run this in a setter for `value`.